packages/babel-plugin-transform-destructuring/src/index.ts

```diff
@@ -80,7 +80,9 @@ export default declare((api, options) => {
           }
 
           statementBody.unshift(
-            t.expressionStatement(t.assignmentExpression("=", left, temp)),
+            t.expressionStatement(
+              t.assignmentExpression("=", left, t.cloneNode(temp)),
```

Comment:
`temp` has been reused when creating `node.left`. I am surprised that `babel-check-duplicated-nodes` does not report this issue. However, it reports when I happen to delete
```
if (statementBody.length === 0 && path.isCompletionRecord()) {
            statementBody.unshift(
              t.expressionStatement(scope.buildUndefinedNode()),
            );
          }
```